ClamavPlugin.inc.php

```diff
@@ -327,51 +331,52 @@ function clamscanHandleUpload($hookName, $args) {
 		//scan for viruses if file exists
 		if (null !== $file) {
 			$useSocket = $this->getSetting(CONTEXT_SITE, 'clamavUseSocket');
-			//Daemon
-			if ($useSocket === true) {
-				try {
+			
+			try {
+				if ($useSocket === true) {
 					$message = $this->_clamDaemonFile($uploadedFile);
-					//safe, continue with submission
-					if ($message == false) {
-						return false;
-					}
 				}
-				catch (ClamScanFailureException $e){
-					//couldn't scan, but ClamAV settings are permissive, continue with submission
-					if ( $this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')==="UNSCANNED_ALLOW") {
-						return false;
-					}
-				}
-			//clamscan executable	
-			} else {
-				try {
+				else {
 					$message = $this->_clamscanFile($uploadedFile);
-					//safe, continue with submission
-					if ($message == false) {
+				}
+					//No viruses found! Continue with submission
+					if ($message === false) {
 						return false;
 					}
-				}
-				catch (ClamScanFailureException $e){
-					//couldn't scan, but ClamAV settings are permissive, continue with submission
-					if ( $this->getSetting(CONTEXT_SITE, 'allowUnscannedFiles')==="UNSCANNED_ALLOW") {
+					//ClamAV reported a virus or failed to complete the scan
+					else {
+						//Prepare to notify the user and halt the upload process
+						$rejectionMessage = ["threatname"=>$message];
+						import('lib.pkp.classes.validation.ValidatorFactory');
+						$schemaService = Services::get('schema');
+						$validator = \ValidatorFactory::make(
```

Comment:
This validator needs to be defined outside of the `try {}` in order to be available in the `catch () {}` here: https://github.com/ulsdevteam/pkp-clamav/pull/9/files/8806cabacc1b06a7736a9276622b4d2d838e756c..de4a01f0218f4d07c79985694b0c454cd583c1e5#diff-87d66e8fad851cdad3963423f16f4937b22cd3bb055bef21cb5f9ad8702e2e8fR374

Reply:
It was complaining about $rejectionMessage not existing, so I just made it an empty array until it gets populated later.